### PR TITLE
Add async/await to unit tests; modify server rendering tests to use async/await.

### PR DESCRIPTION
**Pull request by aickin**
Following on to #9055, this PR adds the ability to use async-await to unit tests, and it changes some of the existing server rendering tests to use async-await.

At @spicyj's suggestion in #9055, the transform is only applied to files in the `__tests__` folders to avoid accidentally transforming shipping product code.

### CODE REVIEW

File: scripts/jest/preprocessor.js [hunk 1]
```diff
@@ -54,11 +55,17 @@ module.exports = {
       !filePath.match(/\/node_modules\//) &&
       !filePath.match(/\/third_party\//)
     ) {
+      // for test files, we also apply the async-await transform, but we want to
+      // make sure we don't accidentally apply that transform to product code.
+      var isTestFile = !!filePath.match(/__tests__/);
```
Comment (spicyj) [inline scripts/jest/preprocessor.js]: Can we do `.match(/\/__tests__\//)`?

File: src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js [hunk 2]
```diff
@@ -19,70 +19,70 @@ let ReactDOMServer;
...
+async function serverRender(reactElement, errorCount = 0) {
+  const markup = await expectErrors(
     () => Promise.resolve(ReactDOMServer.renderToString(reactElement)),
-    errorCount)
-  .then((markup) => {
-    var domElement = document.createElement('div');
-    domElement.innerHTML = markup;
-    return domElement.firstChild;
-  });
+    errorCount);
```
Comment (spicyj) [inline src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js]: nit: fb style is to have this closing ); on the next line, unindented

**Comment by aickin** [comment]
Thanks for the review; addressed in ec62549. Now crossing my fingers on CI.

**Comment by aickin** [comment]
Looks like CI has passed, and I think I've addressed all comments in the review. Can I humbly give this PR a nudge?

**Comment by spicyj** [comment]
Yes of course! I don't get notified when CI runs :(

**Pull request merged**